File: Source/UI/FloatingNames.cs
using System;
using WCSharp.Api;

public class FloatingNameTag
{
    private const float NAME_TAG_HEIGHT = 0.015f;
    private const float NAME_TAG_UPDATE_INTERVAL = 0.03f;
    private AchesTimers NamePosUpdater;
    private readonly IFloatingTags Owner;
    public texttag NameTag;

    public FloatingNameTag(IFloatingTags owner)
    {
        Owner = owner;
        NameTag = texttag.Create();
        Initialize();
    }

    public void Initialize()
    {
        NamePosUpdater = ObjectPool<AchesTimers>.GetEmptyObject();
        SetNameTagAttributes();
        NamePosTimer();
    }

    public void Dispose()
    {
        NameTag?.SetVisibility(false);
        NameTag?.Dispose();
        NamePosUpdater?.Dispose();
    }

    private void SetNameTagAttributes()
    {
        NameTag.SetText(Owner.Name, NAME_TAG_HEIGHT);
        NameTag.SetPermanent(true);
        NameTag.SetColor(114, 188, 212, 255);
        NameTag.SetVisibility(true);
    }

    private void NamePosTimer()
    {
        try
        {
            NamePosUpdater.Timer.Start(NAME_TAG_UPDATE_INTERVAL, true, () =>
            {
                UpdateNameTag();
                Blizzard.SetCameraQuickPositionForPlayer(Owner.Player, Owner.Unit.X, Owner.Unit.Y);
            });
        }
        catch (Exception e)
        {
            Logger.Warning($"Error in FloatingNameTag.NamePosTimer: {e.Message}");
        }
    }

    private void UpdateNameTag() => NameTag.SetPosition(Owner.Unit, NAME_TAG_HEIGHT);

    public static void ShowAllNameTags(player Player, bool shown)
    {
        if (!Player.IsLocal) return;
        for(int i = 0; i < Globals.ALL_KITTIES_LIST.Count; i++)
        {
            var k = Globals.ALL_KITTIES_LIST[i];
            k.NameTag.NameTag.SetVisibility(shown);
        }
    }
}

File: src/UI/FloatingNames.ts
import { Kitty } from 'src/Game/Entities/Kitty/Kitty'
import { Globals } from 'src/Global/Globals'
import { AchesTimers, createAchesTimer } from 'src/Utility/MemoryHandler/AchesTimers'
import { MapPlayer, TextTag } from 'w3ts'

export class FloatingNameTag {
    private NAME_TAG_HEIGHT: number = 0.015
    private NAME_TAG_UPDATE_INTERVAL: number = 0.03
    private NamePosUpdater: AchesTimers
    public Kitty: Kitty
    public NameTag: TextTag

    constructor(kitty: Kitty) {
        this.Kitty = kitty
        this.NameTag = TextTag.create()!
        this.Initialize()
    }

    public Initialize() {
        this.NamePosUpdater = createAchesTimer()
        this.SetNameTagAttributes()
        this.NamePosTimer()
    }

    public dispose() {
        this.NameTag?.setVisible(false)
        this.NameTag?.destroy()
        this.NamePosUpdater?.dispose()
    }

    private SetNameTagAttributes() {
        this.NameTag.setText(this.Kitty.name, this.NAME_TAG_HEIGHT)
        this.NameTag.setPermanent(true)
        this.NameTag.setColor(114, 188, 212, 255)
        this.NameTag.setVisible(true)
    }

    private NamePosTimer() {
        this.NamePosUpdater.Timer.start(this.NAME_TAG_UPDATE_INTERVAL, true, () => {
            this.UpdateNameTag()
            SetCameraQuickPositionForPlayer(this.Kitty.Player.handle, this.Kitty.Unit.x, this.Kitty.Unit.y)
        })
    }

    private UpdateNameTag() {
        return this.NameTag.setPosUnit(this.Kitty.Unit, this.NAME_TAG_HEIGHT)
    }

    public static ShowAllNameTags(Player: MapPlayer, shown: boolean) {
        if (!Player.isLocal()) return
        for (let i: number = 0; i < Globals.ALL_KITTIES_LIST.length; i++) {
            let k = Globals.ALL_KITTIES_LIST[i]
            k.NameTag.NameTag.setVisible(shown)
        }
    }
}
